modules/launcher/src/main/java/org/mule/runtime/module/launcher/service/LifecycleFilterServiceProxy.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.module.launcher.service;
+
+import static org.mule.runtime.core.util.Preconditions.checkArgument;
+import static org.mule.runtime.module.launcher.util.ReflectionUtils.findImplementedInterfaces;
+import org.mule.runtime.api.service.Service;
+import org.mule.runtime.core.api.lifecycle.Startable;
+import org.mule.runtime.core.api.lifecycle.Stoppable;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+
+/**
+ * Proxies a {@link Service} instance to filter invocations of lifecycle methods from
+ * {@link Startable} and {@link Stoppable} interfaces.
+ */
+public class LifecycleFilterServiceProxy implements InvocationHandler
```

Comment:
I have concerns regarding the need for this class. I think that interfaces Startable or Stoppable are meant to be define as part of the Lifecycle API of the MuleContext. A class that will not participate, or should not participate in the lifecycle of the MuleContext should not be Startable or Stoppable. 

Anyway, if we move this logic to the SpringMuleContextServiceConfigurator class, then I think you can define a subclass of ConstantFactoryBean that do not delegates the start/stop method to the service.